src/libNetwork/P2PComm.cpp

```diff
@@ -373,9 +376,25 @@ void SendJobPeers<T>::DoSend() {
       continue;
     }
 
-    SendMessageCore(peer, m_message, m_startbyte, m_hash);
+    // Launch the thread.
+    auto hThread = make_shared<std::future<void>>(
+        std::async(std::launch::async, SendMessageCore, peer, m_message,
+                   m_startbyte, m_hash));
+
+    // Park the thread if message sending taking more than expected
+    auto status = hThread->wait_for(std::chrono::seconds(5));
```

Comment:
If I recall correctly, messages that go beyond a few MB can potentially take (maybe) more than 5 seconds. Maybe we can add this duration to constants.xml instead so we can adjust easily if necessary.